### PR TITLE
Split "edit user" page and move top to GOV.UK Design System

### PR DESCRIPTION
**Pull request by floehopper**
Trello: https://trello.com/c/uZD2I9dj

This splits out the permissions-related functionality from the "edit user" page into a separate "manage permissions" page and then changes the "edit user" page to use the GOV.UK Design System. This is a culmination of a bunch of previous PRs that split out separate pages for various actions an admin can perform on another user, e.g. #2497 & #2509. This means that the "edit user" page no longer contains any forms, but it's now more like a "show user" page which displays various attributes of the user along with a bunch of links to pages for specific actions.

Although there are a *lot* of commits in this PR, almost all of them are small/mechanical preparatory refactorings intended to make the actual changeover relatively straightforward. The key commits are probably:

* [Split manage permissions page out of edit user page ](https://github.com/alphagov/signon/pull/2567/commits/5678e933ddec577fc7e3ad0e1b4bfa580673ec27)
* [Move edit user page to use GOV.UK Design System 🎉](https://github.com/alphagov/signon/pull/2567/commits/4590bb2de22c5e55f4344acc090ad475a1f26031)

And then there are a few commits at the end which make small copy improvements suggested by Etain, Joseph et al, and some tweaks to make things a bit more consistent. In a bid to keep this PR as small (!) as possible, I haven't implemented the copy changes which would've meant slightly more pervasive changes to keep everything consistent.

### Old "edit user" page
Screenshot 2023-12-04 at 16 02 13

### New "edit user" page
Screenshot 2023-12-04 at 15 45 19

### New "manage permissions" page
Screenshot 2023-12-04 at 15 47 29

### Old "suspend user" page
Screenshot 2023-12-04 at 16 02 33

### New "suspend user" page
Screenshot 2023-12-04 at 15 45 40

### Old "access log" page
Screenshot 2023-12-04 at 16 02 51

### New "access log" page
Screenshot 2023-12-04 at 15 46 51

### CODE REVIEW

File: test/helpers/users_helper_test.rb [hunk 1]
```diff
@@ -171,4 +171,35 @@ class UsersHelperTest < ActionView::TestCase
...
+  context "#link_to_resend_invitation" do
+    context "when current user has permission to resend signup email to a user" do
+      setup do
+        stubs(:policy).returns(stub("policy", resend_invitation?: true))
+      end
+
+      should "returns link to resend signup email to user for invited user" do
+        user = build(:invited_user, id: 123)
+        html = link_to_resend_invitation(user)
+        node = Nokogiri::HTML5.fragment(html)
```
Comment (chrislo) [inline test/helpers/users_helper_test.rb:282]: 😎

File: test/support/managing_two_sv_helpers.rb [hunk 2]
```diff
@@ -19,7 +19,7 @@ def assert_user_access_log_contains_messages(user, messages)
   end
 
   def mandate_2sv_for_exempted_user
```
Comment (chrislo) [inline test/support/managing_two_sv_helpers.rb:21]: Minor: this method name could change to `turn_on_2sv_for_exempted_user` to match the text.

File: test/controllers/users_controller_test.rb [hunk 3]
```diff
@@ -338,7 +338,7 @@ class UsersControllerTest < ActionController::TestCase
 
         get :edit, params: { id: user }
 
-        assert_select "a[href='#{edit_user_two_step_verification_mandation_path(user)}']", text: "Mandate 2-step verification for this user"
+        assert_select "a[href='#{edit_user_two_step_verification_mandation_path(user)}']", text: "Turn on 2-step verification for this user"
```
Comment (chrislo) [inline test/controllers/users_controller_test.rb:341]: I half wonder if this route name should change to remove the word "mandate". But I don't think `edit_user_two_step_verification_turn_on_path` reads particularly well, so maybe we can leave it.

**Review by chrislo — APPROVED** [review]
This all looks good to me James! The preparatory commits made it really easy to review and I think the final design-system change is much clearer as a result. I've made a couple of small comments, but nothing that should block this being merged.

File: test/controllers/users_controller_test.rb [hunk 4]
```diff
@@ -396,6 +371,155 @@ class UsersControllerTest < ActionController::TestCase
...
+    end
+
+    context "signed in as Superadmin user" do
+      setup do
+        @superadmin = create(:superadmin_user)
+        sign_in @superadmin
+      end
+
+      should "display a link to change the user's role" do
+        user = create(:user, role: Roles::Normal.role_name)
```
[the remark this answers is not in the document]
Comment (floehopper) [reply test/controllers/users_controller_test.rb]: Ah, nice - I hadn't seen that factory!

File: test/support/managing_two_sv_helpers.rb (hunk 2 above)
Comment (floehopper) [reply test/support/managing_two_sv_helpers.rb:21]: I'd prefer to leave this method name unchanged. I think "mandate" is a more accurate description of what this is doing and the code doesn't always have to mirror the copy exactly.

File: test/controllers/users_controller_test.rb (hunk 3 above)
Comment (floehopper) [reply test/controllers/users_controller_test.rb:341]: As before, I think "mandate" is a more accurate description of what this functionality does and I don't mind that it doesn't match the copy exactly.

File: test/controllers/users_controller_test.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (floehopper) [reply test/controllers/users_controller_test.rb]: Addressed in https://github.com/alphagov/signon/pull/2567/commits/3dfa35bb61f30b41a05146007ce3b38ba14a5d41

File: test/support/managing_two_sv_helpers.rb (hunk 2 above)
Comment (chrislo) [reply test/support/managing_two_sv_helpers.rb:21]: 👍

**Pull request merged**